File: MondayTask/TuesdayTask/ComplexNumber.cs
using System;
using System.Text;

namespace TuesdayTask
{
    /// <summary>
    /// Class realizes complex number and its properties.
    /// </summary>
    public class ComplexNumber
    {
        public double RealPart { get; set; }
        public double ImaginaryPart { get; set; }
        public ComplexNumber()
        {
        }
        /// <summary>
        /// Constructor with two parametrs.
        /// </summary>
        /// <param name="realPart">Real part</param>
        /// <param name="imaginaryPart">Imaginary part</param>
        public ComplexNumber(double realPart, double imaginaryPart)
        {
            RealPart = realPart;
            ImaginaryPart = imaginaryPart;
        }
        /// <summary>
        /// Copy constructor.
        /// </summary>
        /// <param name="newCompNumb">Object that should be copied.</param>
        public ComplexNumber(ComplexNumber newCompNumb)
        {
            RealPart = newCompNumb.RealPart;
            ImaginaryPart = newCompNumb.ImaginaryPart;
        }
        /// <summary>
        /// Addition of two complex numbers.
        /// Generates OwerflowException.
        /// </summary>
        /// <param name="firstNumb">First complex number.</param>
        /// <param name="secondNumb">Second complex number</param>
        /// <returns>Result of addition in new ComplexNumber object.</returns>
        public static ComplexNumber operator + (ComplexNumber firstNumb, ComplexNumber secondNumb)
        {
            checked
            {
                return new ComplexNumber(firstNumb.RealPart + secondNumb.RealPart,
                    firstNumb.ImaginaryPart + secondNumb.ImaginaryPart);
            }
        }
        /// <summary>
        /// Subtracts two complex numbers.
        /// Generates OwerflowException.
        /// </summary>
        /// <param name="firstNumb">First complex number.</param>
        /// <param name="secondNumb">Second complex number</param>
        /// <returns>Result of substraction in new Complex Number object</returns>
        public static ComplexNumber operator - (ComplexNumber firstNumb, ComplexNumber secondNumb)
        {
            checked
            {
                ComplexNumber newNumb = new ComplexNumber();
                newNumb.RealPart = firstNumb.RealPart - secondNumb.RealPart;
                newNumb.ImaginaryPart = firstNumb.ImaginaryPart - secondNumb.ImaginaryPart;
                return newNumb;
            }
        }
        /// <summary>
        /// Multiplicates two complex numbers.
        /// </summary>
        /// <param name="firstNumb">First complex number.</param>
        /// <param name="secondNumb">Second complex number</param>
        /// <returns>New object of complex number that has value of multiplication.</returns>
        public static ComplexNumber operator * (ComplexNumber firstNumb, ComplexNumber secondNumb)
        {
            ComplexNumber newNumb = new ComplexNumber();
            newNumb.RealPart = (firstNumb.RealPart * secondNumb.RealPart)
                - (firstNumb.ImaginaryPart * secondNumb.ImaginaryPart);
            newNumb.ImaginaryPart = (firstNumb.ImaginaryPart * secondNumb.RealPart)
                + (firstNumb.RealPart * secondNumb.ImaginaryPart);
            return newNumb;
        }
        /// <summary>
        /// Devides two complex numbers.
        /// Throw ArgumentException if the calculated values are NAN.
        /// </summary>
        /// <param name="firstNumb">Dividend</param>
        /// <param name="secondNumb">Quotient</param>
        /// <returns>New object of complex number that has value of division.</returns>
        public static ComplexNumber operator /(ComplexNumber firstNumb, ComplexNumber secondNumb)
        {
            ComplexNumber newNumb = new ComplexNumber();
            newNumb.RealPart = ((firstNumb.RealPart * secondNumb.RealPart)
                + (firstNumb.ImaginaryPart * secondNumb.ImaginaryPart))
                / ((secondNumb.RealPart * secondNumb.RealPart) 
                + (secondNumb.ImaginaryPart * secondNumb.ImaginaryPart));

            newNumb.ImaginaryPart = ((firstNumb.ImaginaryPart * secondNumb.RealPart)
                - (firstNumb.RealPart * secondNumb.ImaginaryPart))
                / ((secondNumb.RealPart * secondNumb.RealPart)
                + (secondNumb.ImaginaryPart * secondNumb.ImaginaryPart));
            if (Double.IsNaN(newNumb.RealPart) || Double.IsNaN(newNumb.ImaginaryPart))
            {
                throw new ArgumentException();
            }
            return newNumb;
        }
        /// <summary>
        /// Compares this instance to a complex number numb.
        /// Return the integer that indicates whether this instance 
        /// is less than, equal to, greater than the value of complex number.
        /// </summary>
        /// <param name="numb">First complex number.</param>
        /// <param name="secondNumb">Second complex number</param>
        /// <returns>-1 - is less.
        /// 0 - equals
        /// 1 - is greater.
        /// </returns>
        public int CompareTo(ComplexNumber numb)
        {
            return GetAbsoluteValue(this).CompareTo(GetAbsoluteValue(numb));
        }
        /// <summary>
        /// Gets Absolute value of the complex number.
        /// </summary>
        /// <param name="numb">Complex number</param>
        /// <returns>Double value of complex number absolute value.</returns>
        private double GetAbsoluteValue(ComplexNumber numb)
        {
            return Math.Sqrt((numb.RealPart * numb.RealPart)
                + (numb.ImaginaryPart * numb.ImaginaryPart));
        }
        /// <summary>
        /// Returns string value of complex number.
        /// </summary>
        /// <returns>String of complex number value.</returns>
         public string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append(this.RealPart);
            sb.Append(" + ");
            sb.Append(this.ImaginaryPart);
            sb.Append("i");
            return sb.ToString();
        }
        /// <summary>
        /// Outputs object value in string.
        /// </summary>
        public void ToDisplay()
        {
            Console.WriteLine(this.ToString());
        }
    }
}

File: ComplexNumber/src/main/java/ComplexNumber.java
/**
 * This class discribes complex number, has methods for add, multiple, division,
 * substrat, compare and output on the screen.
 */
public class ComplexNumber {
  private double real;
  private double imaginary;

  public double getReal() {
    return this.real;
  }
  public double getImaginary() {
    return this.real;
  }
  public void setReal() {
    this.real = real;
  }
  public void setImaginary() {
    this.imaginary = imaginary;
  }

  public ComplexNumber() {
  }

  public ComplexNumber(double real, double imaginary) {
    this.real = real;
    this.imaginary = imaginary;
  }

  /**
   * This constructor make new object, that is copy of object from param.
   * @param complexNumber - object, that will be copied in new object.
   */
  public ComplexNumber(ComplexNumber complexNumber) {
    real = complexNumber.real;
    imaginary = complexNumber.imaginary;
  }

  /**
   * This method add one complex number to another.
   * @param first - complex number.
   * @param second - complex number.
   * @return - complex number, that is result of add first complex number to second.
   */
  public ComplexNumber add(ComplexNumber first, ComplexNumber second) {
    return new ComplexNumber(first.real + second.real, first.imaginary + second.imaginary);
  }

  /**
   * This method substract one complex number to another.
   * @param first - complex number.
   * @param second - complex number.
   * @return - complex number, that is result of substract first complex number to another.
   */
  public ComplexNumber substract(ComplexNumber first, ComplexNumber second) {
    return new ComplexNumber(first.real - second.real, first.imaginary - second.imaginary);
  }

  /**
   * This method multiple one complex number to another.
   * @param first - complex number.
   * @param second - complex number.
   * @return - complex number, that is result of multiple first complex number to another.
   */
  public ComplexNumber multiple(ComplexNumber first, ComplexNumber second) {
    return new ComplexNumber(first.real * second.real, first.imaginary * second.imaginary);
  }

  /**
   * This method division one complex number to another.
   * @param first - complex number.
   * @param second - complex number.
   * @return - complex number, that is result of division  first complex number to another.
   */
  public ComplexNumber division(ComplexNumber first, ComplexNumber second) {
    return new ComplexNumber((first.real * second.real + first.imaginary * second.imaginary) /
                              (second.real * second.real + second.imaginary * second.imaginary),
                              (first.imaginary * second.real - first.real * second.imaginary) /
                              (second.real * second.real + second.imaginary * second.imaginary));
  }

  /**
   * This method compare one complex number to another.
   * @param first - complex number.
   * @param second - complex number.
   * @return - true if first and second are equals, false in another case.
   */
  public boolean equals(ComplexNumber first, ComplexNumber second) {
    int tempResult = Double.compare((Math.sqrt(first.real * first.real + first.imaginary + first.imaginary)),
                    Math.sqrt(second.real * second.real + second.imaginary * second.imaginary));
    return (tempResult == 0);
  }

  /**
   * This number output complex number on the screen in typical vision.
   * @param complexNumber - complex number.
   */
  public void outputOnScreen(ComplexNumber complexNumber) {
    System.out.println(complexNumber.real + "i" + complexNumber.imaginary);
  }
}
